vividus-plugin-applitools/src/main/java/org/vividus/visual/eyes/converter/ExamplesTableToApplitoolsVisualChecksConverter.java

```diff
@@ -0,0 +1,226 @@
+    private static final String SERVER_URI_OPTION = "serverUri";
+    private static final String APP_NAME_OPTION = "appName";
+    private static final String BATCH_NAME_OPTION = "batchName";
+    private static final String BASELINE_ENV_NAME_OPTION = "baselineEnvName";
+    private static final String ELEMENTS_TO_IGNORE_OPTION = "elementsToIgnore";
+    private static final String AREAS_TO_IGNORE_OPTION = "areasToIgnore";
+
+    private static final List<String> SUPPORTED_OPTIONS = List.of(BASELINE_NAME_OPTION, ACTION_OPTION,
+            EXECUTE_API_KEY_OPTION, READ_API_KEY_OPTION, HOST_APP_OPTION, HOST_OS_OPTION, VIEWPORT_SIZE_OPTION,
+            MATCH_LEVEL_OPTION, SERVER_URI_OPTION, APP_NAME_OPTION, BATCH_NAME_OPTION, BASELINE_ENV_NAME_OPTION,
+            ELEMENTS_TO_IGNORE_OPTION, AREAS_TO_IGNORE_OPTION);
+
+    private String executeApiKey;
+    private String readApiKey;
+    private String hostApp;
+    private String hostOS;
+    private Dimension viewportSize;
+    private MatchLevel matchLevel;
+    private URI serverUri;
+    private String appName = "Application";
+    private String baselineEnvName;
+
+    private final Map<String, BatchInfo> batchStorage = new ConcurrentHashMap<>();
+
+    private final ScreenshotParametersFactory<ScreenshotConfiguration> screenshotParametersFactory;
+    private final BaselineIndexer baselineIndexer;
+
+    public ExamplesTableToApplitoolsVisualChecksConverter(
+            ScreenshotParametersFactory<ScreenshotConfiguration> screenshotParametersFactory,
+            BaselineIndexer baselineIndexer)
+    {
+        this.screenshotParametersFactory = screenshotParametersFactory;
+        this.baselineIndexer = baselineIndexer;
+    }
+
+    @Override
+    public List<ApplitoolsVisualCheck> convertValue(ExamplesTable table, Type type)
+    {
+        List<String> unsupportedOptions = table.getHeaders().stream()
+                .filter(option -> !SUPPORTED_OPTIONS.contains(option))
+                .collect(Collectors.toList());
+        Validate.isTrue(unsupportedOptions.isEmpty(), "Unknown Applitools configuration options: %s",
+                String.join(", ", unsupportedOptions));
+
+        return table.getRowsAsParameters(true).stream().map(params ->
+        {
+            String batchName = params.valueAs(BATCH_NAME_OPTION, String.class);
+            String baselineName = params.valueAs(BASELINE_NAME_OPTION, String.class);
+            VisualActionType action = params.valueAs(ACTION_OPTION, VisualActionType.class);
+
+            ApplitoolsVisualCheck check = createCheck(batchName, baselineName, action);
+            Type targetType = new TypeLiteral<Set<Locator>>() { }.value;
+            configureCommons(
+                check,
+                params.valueAs(READ_API_KEY_OPTION, String.class, readApiKey),
+                params.valueAs(ELEMENTS_TO_IGNORE_OPTION, targetType, Set.of()),
+                params.valueAs(AREAS_TO_IGNORE_OPTION, targetType, Set.of())
+            );
```

Comment:
`createCheck` and `configureCommons` can be merged into a single method

Reply:
I wanted to separate them logically to not have one method with so many params